### ISSUE TITLE
Generator loss keeps increasing in Map dataset using default parameters

### ISSUE THREAD

**Issue by boyurmit** [affinelayer/pix2pix-tensorflow#45]
Hi, I was running this code on Map dataset by the default parameters. However, the _generator loss_ keeps increasing while the _discriminator_ and _L1 loss_ keep decreasing. The plots are as follows, 
![gen_loss](https://user-images.githubusercontent.com/15166862/27908887-5b3c1bf2-6291-11e7-9796-9b8e2554c41c.PNG)
![gen_l1_loss](https://user-images.githubusercontent.com/15166862/27908889-613e4c14-6291-11e7-9d72-bb22ae8d2a7f.PNG)
![dis_loss](https://user-images.githubusercontent.com/15166862/27908892-63226f2e-6291-11e7-8ec9-b088747aa2ac.PNG)

I wonder if this is the same case in your experiment. Since for a GAN model, I expect the _generator loss_ should keep decreasing.

**Comment by julien2512** [comment]
You shall try to adjust the Formula gen_loss + N * L1_loss decreasing N.

Your samples might also be too easy for the L1 discriminator.

Julien.

**Comment by boyurmit** [comment]
@julien2512 Thanks for your reply. I was using the provided dataset _Map_ in the paper and the same parameters. I don't know whether they get the same result as mine. I've tried decreasing N to, say 0.1, but the _gen_loss_ still gets increasing. I may try smaller N later.

**Comment by julien2512** [comment]
@boyurmit You might also notice the sample NN is made of 57 000 000 parameters and more ! Deep learning is about seeking a solution in such a wide space. If you can't scale up, or adjust the NN, or using more data, the only path is to wait !

**Comment by boyurmit** [comment]
@julien2512 I think you still didn't answer my question. I want to reproduce the experimental result in the paper by the same dataset and same parameters. If I get this questionable result but the paper doesn't, then there is a problem.

**Comment by julien2512** [comment]
@boyurmit Sorry for my misunderstanding. I got the same results as you. But
the paper is about to compare methods and shows GAN+L1 is better than only
GAN nor only L1, right ? The discriminator and generator are challenging
and there is an actual influence of that on the gradient descent.
On my side, I've got other dataset where GAN used to go down for a part of
the dataset, and go up again for the other part.

2017-07-08 2:35 GMT+02:00 boyurmit <notifications@github.com>:

**Comment by boyurmit** [comment]
All right, that makes sense. The dataset really do have influence on the training. Thanks @julien2512.

**Comment by groot-1313** [comment]
The paper states:

> As suggested in the original GAN paper, rather than training G to minimize log(1 − D(x, G(x, z)), we instead train to maximize log D(x, G(x, z)).

**Comment by julien2512** [comment]
See my answer on #119
